video/out/meson.build

```diff
@@ -1,65 +1,52 @@
-wl_protocol_dir = wayland['deps'][2].get_variable(pkgconfig: 'pkgdatadir', internal: 'pkgdatadir')
-protocols = [[wl_protocol_dir, 'stable/presentation-time/presentation-time.xml'],
-             [wl_protocol_dir, 'stable/viewporter/viewporter.xml'],
-             [wl_protocol_dir, 'stable/xdg-shell/xdg-shell.xml'],
-             [wl_protocol_dir, 'unstable/idle-inhibit/idle-inhibit-unstable-v1.xml'],
-             [wl_protocol_dir, 'unstable/text-input/text-input-unstable-v3.xml'],
-             [wl_protocol_dir, 'unstable/linux-dmabuf/linux-dmabuf-unstable-v1.xml'],
-             [wl_protocol_dir, 'unstable/xdg-decoration/xdg-decoration-unstable-v1.xml'],
-             [wl_protocol_dir, 'staging/content-type/content-type-v1.xml'],
-             [wl_protocol_dir, 'staging/fractional-scale/fractional-scale-v1.xml'],
-             [wl_protocol_dir, 'staging/single-pixel-buffer/single-pixel-buffer-v1.xml'],
-             [wl_protocol_dir, 'staging/xdg-activation/xdg-activation-v1.xml']]
-wl_protocols_source = []
-wl_protocols_headers = []
+protocols = [wl_mod.find_protocol('presentation-time'),
+             wl_mod.find_protocol('viewporter'),
+             wl_mod.find_protocol('xdg-shell'),
+             wl_mod.find_protocol('idle-inhibit', state: 'unstable', version: 1),
+             wl_mod.find_protocol('text-input', state: 'unstable', version: 3),
+             wl_mod.find_protocol('linux-dmabuf', state: 'unstable', version: 1),
+             wl_mod.find_protocol('xdg-decoration', state: 'unstable', version: 1),
+             wl_mod.find_protocol('content-type', state: 'staging', version: 1),
+             wl_mod.find_protocol('fractional-scale', state: 'staging', version: 1),
+             wl_mod.find_protocol('single-pixel-buffer', state: 'staging', version: 1),
+             wl_mod.find_protocol('xdg-activation', state: 'staging', version: 1)
+]
 
 foreach v: ['1.32', '1.38', '1.39', '1.41', '1.44']
     features += {'wayland-protocols-' + v.replace('.', '-'):
         wayland['deps'][2].version().version_compare('>=' + v)}
 endforeach
 
 if features['wayland-protocols-1-32']
-    protocols += [[wl_protocol_dir, 'staging/cursor-shape/cursor-shape-v1.xml'],
-                  [wl_protocol_dir, 'unstable/tablet/tablet-unstable-v2.xml']] # required by cursor-shape
+    protocols += [wl_mod.find_protocol('cursor-shape', state: 'staging', version: 1),
+                  wl_mod.find_protocol('tablet', state: 'unstable', version: 2)] # required by cursor-shape
 endif
 
-
 if features['wayland-protocols-1-38']
```

Comment:
Why manual version checking? Shouldn't it now be handled by `find_protocol` and if not found it should return none value or something? Same as find_dependency. Else using this module is pointless, if we need to do all the checking ourselfs.